packages/memory/src/tools/working-memory.ts

```
@@ -29,27 +29,23 @@ export const updateWorkingMemoryTool = (memoryConfig?: MemoryConfig) => {
-      if (!thread) {
-        thread = await memory.createThread({
-          threadId,
-          resourceId,
-          memoryConfig,
-        });
+      if (!threadId || !memory || !resourceId) {
+        throw new Error('Thread ID, Memory instance, and resourceId are required for working memory updates');
       }
 
-      if (thread.resourceId && thread.resourceId !== resourceId) {
+      if (resourceId && resourceId !== resourceId) {
```

Biome (2.1.2)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)